src/gauge/radial-gauge/radial-gauge.component.ts

```diff
@@ -0,0 +1,266 @@
+    ? this.displayValue 
+    : this.value.toString();
+
+    this.colors = new ColorHelper(this.scheme, this.schemeType, 
+      [this.minValue, this.maxValue], this.customColors);
+
+    this.arcs = this.getArcs();
+  }
+
+  public getTranslate(): string {
+    const twidth = this.width / 2;
+    const theight = this.height / 2;
+    return `translate(${twidth}, ${theight})`;
+  }
+
+  public movePointer(): string {
+    return `rotate(${this.getPointerLocation()})`;
+  }
+
+  public getTextTransform(): string {
+    const scale: number = this.getElementScale(0.004);
+    return `scale(${scale}, ${scale})`;
+  }
+
+  public getPointer(): string {
+    const pointerLine: Line<[number, number]> = line().curve(curveLinear);
+    return pointerLine(this.getPointerData());
+  }
+
+  public getScale(): ScaleLinear<number, number> {
+    return scaleLinear()
+    .range([0, this.getDegreeRange()])
+    .nice()
+    .domain([this.minValue, this.maxValue]);
+  }
+
+  public getArcs(): any {
+    const result = [];
+
+    if (this.segments != null && this.segments.length > 0) {
+      for(const i in this.segments) {
+        const arc = {
+          startAngle: this.getSegmentAngle(this.segments[i].minValue),
+          endAngle: this.getSegmentAngle(this.segments[i].maxValue),
+          innerRadius: this.dimensions.innerArcRadius,
+          outerRadius: this.dimensions.outerArcRadius,
+          color: this.getValueOr(this.segments[i].color, 
+            this.colors.getColor(this.segments[i].minValue))
+        };
+        result.push(arc);
+      }
+    } else {
+      for(let i = 0; i < this.majorTicks; i++) {
+        const arc = {
+          startAngle: this.getTickAngle(i),
+          endAngle: this.getTickAngle(i + 1),
+          innerRadius: this.dimensions.innerArcRadius,
+          outerRadius: this.dimensions.outerArcRadius,
+          color: this.colors.getColor(this.ticks[i])
+        };
+        result.push(arc);
+      }
+    }
+    return result;
+  }
+
+  public onArcClick(event: any): void {
+    this.click.emit(event);
+    this.arcClick.emit(event);
+  }
+
+  public onArcActivate(event: any): void {
+    this.mouseEnter.emit(event);
+    this.arcMouseEnter.emit(event);
+  }
+
+  public onArcDeactivate(event: any): void {
+    this.mouseLeave.emit(event);
+    this.arcMouseLeave.emit(event);
+  }
+
+  public onLabelClick(event: any): void {
+    this.click.emit(event);
+    this.labelClick.emit(event);
+  }
+
+  public onLabelMouseEnter(event: any): void {
+    this.mouseEnter.emit(event);
+    this.labelMouseEnter.emit(event);
+  }
+
+  public onLabelMouseLeave(event: any): void {
+    this.mouseLeave.emit(event);
+    this.labelMouseLeave.emit(event);
+  }
+
+  public onPointerClick(event: any): void {
+    this.click.emit(event);
+    this.pointerClick.emit(event);
+  }
+
+  public onPointerMouseEnter(event: any): void {
+    this.pointerMouseEnter.emit(event);
+    this.mouseEnter.emit(event);
```

Comment:
Did you want the `this.mouseEnter` event to be fired before the `this.pointerMouseEnter`?